### PR TITLE
Complete routing for Java-to-Dart messages

### PR DESCRIPTION
**Pull request by abarth**
This patch completes the routing for Java-to-Dart platform messages.

### CODE REVIEW

**Comment by abarth** [comment]
@jason-simmons @chinmaygarde

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -172,6 +220,24 @@ void PlatformViewAndroid::HandlePlatformMessage(
   }
 }
 
+void PlatformViewAndroid::HandlePlatformMessageResponse(
+    int response_id,
+    std::vector<char> data) {
+  JNIEnv* env = base::android::AttachCurrentThread();
+  {
```
Comment (chinmaygarde) [inline shell/platform/android/platform_view_android.cc]: This extra scope is unnecessary.

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 2]
```diff
@@ -712,12 +728,23 @@ private boolean handleAccessibilityHoverEvent(MotionEvent event) {
         return true;
     }
 
+    private void dispatchPlatformMessage(String name, String message, MssageReplyCallback callback) {
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/view/FlutterView.java]: typo - "MssageReplyCallback"

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 3]
```diff
@@ -597,17 +602,28 @@ private void handlePlatformMessage(String name, String message, final int respon
...
         nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, null);
     }
 
+    private int mNextResponseId = 1;
+    private final Map<Integer, MessageReplyCallback> mPendingResponses = new HashMap<Integer, MessageReplyCallback>();
+
+    // Called by native to respond to a platform message that we sent.
+    @CalledByNative
+    private void handlePlatformMessageResponse(int responseId, String response) {
+        MessageReplyCallback callback = mPendingResponses.remove(responseId);
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Is this being invoked on the Dart UI thread?
If so, then the callback and the access to the mPendingResponse map may need to be queued to the Android main thread

Comment (abarth) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Yes, I'm missing the thread hop.  Thanks.  :)

**Pull request merged**